src/types/signature.ts

```diff
@@ -0,0 +1,17 @@
+import { OneOf } from './utility';
+
+interface VerificationMaterial {
+  certificates: string[];
+  key: {
+    id?: string;
+    value: string;
+  };
+}
+
+// Result of signing some artifact. Containing the signature and either the
+// signing certificate chain or the public key.
+export type SignatureMaterial = {
+  signature: Buffer;
+} & OneOf<VerificationMaterial>;
```

Comment:
Introduces a new, internal type to carry around the generated signature and information about the certificate/key used to generate that signature.